test/currencyConvertor.ts

```diff
@@ -1,73 +1,196 @@
+    });
+    await hre.network.provider.request({
+      method: "hardhat_setBalance",
+      params: [
+        impersonatedAccount,
+        "0x1000000000000000000",
+      ],
+    });
+    signer = await ethers.getSigner(impersonatedAccount);
 
     currencyConvertor = await deployContract(
-      signers[0],
+      signer,
       CurrencyConvertorArtifact,
+    ) as CurrencyConvertor;
+    zeroExExchangeWrapper = await deployContract(
+      signer,
+      ZeroExExchangeArtifact,
       [
-        starkware.address,
-        usdc.address, // placeholder for USDC
-        156,
+        starkwareContractAddress,
+        usdcAddress,
+        '0x02893294412a4c8f915f75892b395ebbf6859ec246ec365c3b1f56f47c3a0a5d',
       ],
-    );
+    ) as ZeroExExchangeWrapper;
 
-    await usdc.mint(
-      signers[0].address,
-      1000e6,
-    )
+    // get ERC20 contracts
+    usdtTokenContract = new ethers.Contract(
+      usdtTokenAddress,
+      erc20Abi,
+      signer,
+    ) as ERC20;
+    usdcTokenContract = new ethers.Contract(
+      usdcAddress,
+      erc20Abi,
+      signer,
+    ) as ERC20;
 
-    await usdc.approve(currencyConvertor.address, 1e10);
+    // approve ERC20 contracts
+    await usdtTokenContract.approve(
+      currencyConvertor.address,
+      100000000000,
+    );
   });
+
+  after(async () => {
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [impersonatedAccount],
+    });
+  })
+
   describe("deposit", async () => {
-    it("deposit USDT to USDC", async () => {
+    it("deposit USDT as USDC to Starkware", async () => {
+      const zeroExTransaction = await zeroExRequest('100');
+
+      // get old balances
+      const userUsdtBalance: BigNumber = await usdtTokenContract.balanceOf(signer.address);
+      const starkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      )
+
+      await zeroExExchangeWrapper.approveSwap(zeroExTransaction.to, usdtTokenAddress);
+      const tx = await currencyConvertor.deposit(
+        usdtTokenAddress,
+        '100000',
+        zeroExExchangeWrapper.address,
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
+        ethers.utils.getAddress(zeroExTransaction.to),
+        zeroExTransaction.data,
+      );
+
+      const blocks = await tx.wait();
+      const events = _.chain(blocks.events!)
+      .filter((e) => e.event === 'LogConvertedDeposit')
+      .value();
+
+      const event = events[0];
+      expect(event.args?.tokenFromAmount.toString()).to.equal('100000');
+      expect(event.args?.tokenFrom.toLowerCase()).to.equal(usdtTokenAddress);
+
+      // get new balances
+      const newUserUsdtBalance: BigNumber = await usdtTokenContract.balanceOf(signer.address);
+      const newStarkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      )
```

Comment:
nit: Run a linter (missing semicolon)

Reply:
fixed this and another but Id rather not add a whole linter for tests